src/main/java/org/cactoos/list/LimitedIterable.java

```diff
@@ -0,0 +1,40 @@
+package org.cactoos.list;
+
+import java.util.Iterator;
+
+/**
+ * Limited iterable.
+ *
+ * <p>This is a view of an existing iterable containing the given number of its
+ * first elements.</p>
+ *
+ * @author Dusan Rychnovsky (dusan.rychnovsky@gmail.com)
+ * @param <T> Element type
+ * @since 0.4
+ */
+public class LimitedIterable<T> implements Iterable<T> {
+  
+    private final Iterable<T> iterable;
+    private final int limitSize;
+
+    /**
+     * Ctor.
+     *
+     * @param iterable the underlying iterable
+     * @param limitSize the requested number of elements
+     */
+    public LimitedIterable(Iterable<T> iterable, int limitSize) {
```

Comment:
@yegor256 I understand the second part, but which name are you referring to regarding avoiding collisions - the decorated iterable? Please explain. Thanks